logo.js

```diff
@@ -818,21 +872,37 @@ function LogoInterpreter(turtle, stream, savehook)
     // Operate on a copy so the original is not destroyed
     statements = statements.slice();
 
-    var result;
-    while (statements.length) {
-      result = evaluateExpression(statements);
-
-      if (result !== undefined && !options.returnResult) {
-        throw new Error(format(__("Don't know what to do with {result}"), {result: result}));
+    return new Promise(function (resolve, reject) {
+      var lastResult;
+      function runLoop() {
+        while (statements.length) {
+          var result = evaluateExpression(statements);
+          if (isPromise(result)) {
```

Comment:
else? Do we have any expressions that are not returning promises at this point? Internal error?